### DISCUSSION TITLE
Configuration option to configure delay of generated i18n index redirect to reduce waiting time for users

### DISCUSSION THREAD

**Discussion by sceee** [withastro/roadmap#812] [Proposal]
### Body

# Summary

At the moment, when using `i18n`, an `index.html` is generated in the root that redirects to the default language using a http meta refresh after 2 seconds.

This proposal is about making this redirect `delay` configurable.

# Background & Motivation

The redirect delay of 2 seconds (see https://github.com/withastro/astro/blob/d02a3c48a3ce204649d22e17b1e26fb5a6a60bcf/packages/astro/src/core/build/generate.ts#L582 ) causes the user to have to wait for 2 seconds before being redirected to the default language specific index.

I want to shorten it (without making it to be considered a _permanent redirect_ according to the comment in the code mentioned), therefore possibly configure it to 1 second.

This would at least reduce the waiting time by 50%.

# Goals

- Reduce the waiting time for users visiting the website before seeing the website

# Example

Add a new config option
```i18n
export default defineConfig({
  i18n: {
    redirectToDefaultLocaleDelay: 1, <---- This is the new option

    defaultLocale: "de",
    locales: ["de", "en"],
    routing: {
      prefixDefaultLocale: true,
    },
  },
});
```

**Comment by ematipico** [comment]
There's a reason why the delay is two seconds: https://developers.google.com/search/docs/crawling-indexing/301-redirects#metarefresh

> - Instant meta refresh redirect: Triggers as soon as the page is loaded in a browser. Google Search interprets instant meta refresh redirects as permanent redirects.
>
> - Delayed meta refresh redirect: Triggers only after an arbitrary number of seconds set by the site owner. Google Search interprets delayed meta refresh redirects as temporary redirects.

Although I can see why a user would want to have a different delay. I think it makes sense.

**Reply by sceee** [reply]
Yeah but from the Google docs, I only see a difference for
* 0 seconds (=immediate redirect => treated as 301)
* anything more than 0 seconds (= redirect after the delay => treated as 302)

I did not find a reason why it specifically would have to be **2** seconds to become 302 (and why 1 second would not also trigger a 302).

**Reply by ematipico** [reply]
I looked at the final RFC of redirects and I couldn't find a technical reason for that. cc @matthewp , he might have some info

**Reply by matthewp** [reply]
A user that seemed to know what he was talking about suggested the 2 seconds, saying that's what Google considered. I don't recall there being a documented link, but you could look at the RFC again to check.

For this RFC @sceee would you want a global configuration option for all redirects or specifically just for this one?

**Reply by sceee** [reply]
@matthewp alright, then someone probably knows more than I do so the 2 seconds might be worth keeping.
I actually only know this `/` to `/[defaultlang]/` redirect - so I thought just one global configuration would be nice to have.

But as written, as it seems there is a good reason why it's exactly 2 seconds, I'll probably better keep it.
